applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -1693,6 +1775,96 @@ void BaseSolidElement::CalculateAndAddExtForceContribution(
 /***********************************************************************************/
 /***********************************************************************************/
 
+void BaseSolidElement::CalculateLumpedMassMatrix(MatrixType& rMassMatrix)
+{
+    KRATOS_TRY;
+
+    const auto& r_geom = GetGeometry();
+    const auto& r_prop = GetProperties();
+    SizeType dimension = r_geom.WorkingSpaceDimension();
+    SizeType number_of_nodes = r_geom.size();
+    SizeType mat_size = dimension * number_of_nodes;
+
+    rMassMatrix = ZeroMatrix( mat_size, mat_size );
+
+    KRATOS_ERROR_IF_NOT(r_prop.Has( DENSITY )) << "DENSITY has to be provided for the calculation of the MassMatrix!" << std::endl;
+
+    const double density = r_prop[DENSITY];
+    const double thickness = (dimension == 2 && r_prop.Has(THICKNESS)) ? r_prop[THICKNESS] : 1.0;
+
+    // LUMPED MASS MATRIX
+    const double total_mass = GetGeometry().DomainSize() * density * thickness;
+
+    Vector lumping_factors;
+    lumping_factors = GetGeometry().LumpingFactors( lumping_factors );
+
+    for ( IndexType i = 0; i < number_of_nodes; ++i ) {
+        const double temp = lumping_factors[i] * total_mass;
+        for ( IndexType j = 0; j < dimension; ++j ) {
+            IndexType index = i * dimension + j;
+            rMassMatrix( index, index ) = temp;
+        }
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void BaseSolidElement::CalculateLumpedDampingMatrix(
+    MatrixType& rDampingMatrix,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_TRY;
+
+    unsigned int number_of_nodes = GetGeometry().size();
+    unsigned int dimension = GetGeometry().WorkingSpaceDimension();
+
+    // Resizing as needed the LHS
+    unsigned int mat_size = number_of_nodes * dimension;
+
+    if ( rDampingMatrix.size1() != mat_size )
+        rDampingMatrix.resize( mat_size, mat_size, false );
+
+    noalias( rDampingMatrix ) = ZeroMatrix( mat_size, mat_size );
+
+    // 1.-Calculate StiffnessMatrix:
+
+    MatrixType stiffness_matrix( mat_size, mat_size );
+    VectorType residual_vector( mat_size );
+
+    this->CalculateAll(stiffness_matrix, residual_vector, rCurrentProcessInfo, true, false);
+
+    // 2.-Calculate mass matrix:
+    MatrixType mass_matrix( mat_size, mat_size );
+    CalculateLumpedMassMatrix ( mass_matrix );
```

Comment:
this function is one-to-one the other one except this line changed
Why don't you select whether to compute the lumped-MM through the Properties?

Reply:
In explicit is always lumped

Reply:
if you know it is lumped you can avoid using a matrix and use a v ctor to store the diagonal